lib/js-api/src/externals/memory.rs

```diff
@@ -0,0 +1,331 @@
+    /// let p = m.grow(2).unwrap();
+    ///
+    /// assert_eq!(p, Pages(1));
+    /// assert_eq!(m.size(), Pages(3));
+    /// ```
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if memory can't be grown by the specified amount
+    /// of pages.
+    ///
+    /// ```should_panic
+    /// # use wasmer_js::{Memory, MemoryType, Pages, Store, Type, Value, WASM_MAX_PAGES};
+    /// # let store = Store::default();
+    /// #
+    /// let m = Memory::new(&store, MemoryType::new(1, Some(1), false)).unwrap();
+    ///
+    /// // This results in an error: `MemoryError::CouldNotGrow`.
+    /// let s = m.grow(1).unwrap();
+    /// ```
+    pub fn grow<IntoPages>(&self, delta: IntoPages) -> Result<Pages, MemoryError>
+    where
+        IntoPages: Into<Pages>,
+    {
+        let pages = delta.into();
+        let js_memory = self.vm_memory.memory.clone().unchecked_into::<JSMemory>();
+        let new_pages = js_memory.grow(pages.0).map_err(|err| {
+            if err.is_instance_of::<js_sys::RangeError>() {
+                MemoryError::CouldNotGrow {
+                    current: self.size(),
+                    attempted_delta: pages,
+                }
+            } else {
+                MemoryError::Generic(err.as_string().unwrap())
+            }
+        })?;
+        Ok(Pages(new_pages))
+    }
+
+    /// Return a "view" of the currently accessible memory. By
+    /// default, the view is unsynchronized, using regular memory
+    /// accesses. You can force a memory view to use atomic accesses
+    /// by calling the [`MemoryView::atomically`] method.
+    ///
+    /// # Notes:
+    ///
+    /// This method is safe (as in, it won't cause the host to crash or have UB),
+    /// but it doesn't obey rust's rules involving data races, especially concurrent ones.
+    /// Therefore, if this memory is shared between multiple threads, a single memory
+    /// location can be mutated concurrently without synchronization.
+    ///
+    /// # Usage:
+    ///
+    /// ```
+    /// # use wasmer_js::{Memory, MemoryView};
+    /// # use std::{cell::Cell, sync::atomic::Ordering};
+    /// # fn view_memory(memory: Memory) {
+    /// // Without synchronization.
+    /// let view: MemoryView<u8> = memory.view();
+    /// for byte in view[0x1000 .. 0x1010].iter().map(Cell::get) {
+    ///     println!("byte: {}", byte);
+    /// }
+    ///
+    /// // With synchronization.
+    /// let atomic_view = view.atomically();
+    /// for byte in atomic_view[0x1000 .. 0x1010].iter().map(|atom| atom.load(Ordering::SeqCst)) {
+    ///     println!("byte: {}", byte);
+    /// }
+    /// # }
+    /// ```
+    pub fn view<T: ValueType>(&self) -> MemoryView<T> {
+        unimplemented!("The view function is not yet implemented in Wasmer Javascript");
+    }
+
+    /// example view
+    pub fn uint8view(&self) -> js_sys::Uint8Array {
+        js_sys::Uint8Array::new(&self.vm_memory.memory.buffer())
+    }
```

Comment:
`uint8view` is an example here, not a real method it seems like.

However, we can improve the `view` method to produce a `Uint8Array` or simply to provide an `ArrayBuffer`. We could even mark `view` as unimplemented, and provide a `buffer` method. Or just re-use `data_ptr` to provide an `ArrayBuffer`.

I think we should try to make `view` working with `ArrayBuffer`, that would be the best.